### PR TITLE
Check authentication message length wrt to maximum number of pages

### PR DESCRIPTION
**Pull request by cedric-papon**
Fix for  #48

### CODE REVIEW

File: libopendroneid/opendroneid.c [hunk 1]
```diff
@@ -378,10 +378,18 @@ int encodeAuthMessage(ODID_Auth_encoded *outEncoded, ODID_Auth_data *inData)
...
 
+#if (ODID_AUTH_MAX_PAGES <= 11) 
+        if (inData->Length > MAX_AUTH_LENGTH)
+            return ODID_FAIL;
+#else
+        if (inData->LastPageIndex > 10 &&
+            inData->Length != UINT8_MAX)
+            return ODID_FAIL;
+#endif
+
```
Comment (friissoren) [inline libopendroneid/opendroneid.c:392]: I would propose to fix this instead with the following code snippet:
```
#if (MAX_AUTH_LENGTH < UINT8_MAX)
        if (inData->Length > MAX_AUTH_LENGTH)
            return ODID_FAIL;
#endif
```

File: libopendroneid/opendroneid.c [hunk 2]
```diff
@@ -748,9 +756,17 @@ int decodeAuthMessage(ODID_Auth_data *outData, ODID_Auth_encoded *inEncoded)
...
-            inEncoded->page_zero.Length > MAX_AUTH_LENGTH)
+        if (inEncoded->page_zero.LastPageIndex >= ODID_AUTH_MAX_PAGES)
+            return ODID_FAIL;
+
+#if (ODID_AUTH_MAX_PAGES <= 11) 
+        if (inEncoded->page_zero.Length > MAX_AUTH_LENGTH)
+            return ODID_FAIL;
+#else
+        if (inEncoded->page_zero.LastPageIndex > 10 &&
+            inEncoded->page_zero.Length != UINT8_MAX)
```
Comment (friissoren) [inline libopendroneid/opendroneid.c:767]: I would propose to fix this instead with the following code snippet:
```
#if (MAX_AUTH_LENGTH < UINT8_MAX)
        if (inEncoded->page_zero.Length > MAX_AUTH_LENGTH)
            return ODID_FAIL;
#endif
```

File: libopendroneid/opendroneid.c (hunk 1 above)
Comment (cedric-papon) [reply libopendroneid/opendroneid.c:388]: Better. Done.

File: libopendroneid/opendroneid.c (hunk 2 above)
Comment (cedric-papon) [reply libopendroneid/opendroneid.c]: Better. Done.

**Comment by cedric-papon** [comment]
Implemented your proposed changes, thanks.

Should we add "-Wextra" flag in the root CMakeLists.txt ?

BTW I also checked with cppcheck (1.90) and clang-tidy-10 for sanity.

**Comment by friissoren** [comment]
Yes, I definitely think it would be a good idea to add -Wextra to the compile flags. However, there are a few additional warnings in odid_test that should be handled then also. I can take a look at that.

**Comment by friissoren** [comment]
-Wextra flag now added. And a bunch of lint warnings fixed.

**Comment by friissoren** [comment]
-Wextra flag now added. And a bunch of lint warnings fixed.

**Pull request merged**